R/clean_data_helpers.R

```diff
@@ -1,76 +1,147 @@
+#' @return A numeric vector with the proportion of the different types of data
+#'    that were detected within the input vector.
 #' @keywords internal
 #'
-scan_columns <- function(x) {
-  # --- save the variable length ---
-  n_rows <- length(x)
+scan_in_character <- function(x) {
+  # There might be, within a character column, values of type:
+  # character, numeric, date (date or date-time), NA, and logical
+  # In this function, we check the presence of these different types within a
+  # character column.
+
+  # save the variable length
+  initial_length <- length(x)
 
-  # --- get the proportion of NA ---
-  are_na <- round((sum(is.na(x)) / n_rows), 6L)
+  # get the count of missing data (NA)
+  na_count <- sum(is.na(x))
   x      <- x[!is.na(x)]
 
-  # --- get the proportion of numeric values ---
-  tmp         <- suppressWarnings(as.numeric(x))
-  are_numeric <- round((sum(!is.na(tmp)) / n_rows), 6L)
+  # We will check if there is any Date values within the variable by parsing the
+  # values, looking for the ones that fit any of the predefined format.
+  #     When there is one or more Date values, we will convert the variable into
+  # numeric and determine if any of them is a Date (a numeric, which after
+  # conversion to Date, fall within the interval
+  # [50 years back from today's date, today's date]). That way the Date count is
+  # the count of date identified from the parsing + the count of Dates within
+  # the numeric values.
+  #     When there is no Date values identified from the parsing, the variable
+  # is converted into numeric. The numeric count is the sum of numeric values.
+  #     The logical count is the number of TRUE and FALSE written in both lower
+  # and upper case within the variable
+  #     The remaining values will be of type character.
 
-  # --- get the proportion of date values ---
-  x        <- x[which(is.na(tmp))]
-  are_date <- 0L
-  if (!is.null(lubridate::guess_formats(x, c("ymd", "ydm", "dmy", "mdy", "myd",
-                                             "dym", "Ymd", "Ydm", "dmY", "mdY",
-                                             "mYd", "dYm")))) {
-    x <- suppressWarnings(
-      as.Date(
-        lubridate::parse_date_time(
-          x, orders = c("ymd", "ydm", "dmy", "mdy", "myd", "dym", "Ymd", "Ydm",
-                        "dmY", "mdY", "mYd", "dYm")
-        )
+  # parsing the vector, looking for date values
+  tmp <- suppressWarnings(
+    as.Date(
+      lubridate::parse_date_time(
+        x,
+        orders = c("ymd", "ydm", "dmy", "mdy", "myd", "dym", "Ymd", "Ydm",
+                   "dmY", "mdY", "mYd", "dYm")
       )
     )
-    are_date <- round((sum(!is.na(x)) / n_rows), 6L)
+  )
+
+  # getting the date and numeric count as describe above
+  date_count <- numeric_count <- 0L
+  if (sum(!is.na(tmp)) > 0L) {
+    # Setting the first date to 50 years before the current date
+    target_interval <- sort(
+      seq.Date(Sys.Date(), length.out = 2L, by = "-50 years")
+    )
```

Comment:
Only the first value is used thereafter. What about creating a simpler `oldest_possible_date` object which would be a `Date` of length 1?